docs/usage/usage-with-typescript.md

```diff
@@ -28,6 +28,35 @@ const rootReducer = combineReducers({})
 export type RootState = ReturnType<typeof rootReducer>
 ```
 
+If you pass the reducers directly to `configureStore()` and do not define the root reducer explicitly, there is no reference to `rootReducer`. 
+Instead, you can refer to `store.getState`, in order to get the `State` type.
+
+```typescript
+import { configureStore } from '@reduxjs/toolkit'
+import rootReducer from './rootReducer'
+const store = configureStore({
+  reducer: rootReducer
+})
+export type RootState = ReturnType<typeof store.getState>
+```
+
+Note that passing the reducers directly is not recommended when you need to add custom middleware,
+because this can cause a circular reference in TypeScript:
+
+```typescript
+import { configureStore } from '@reduxjs/toolkit'
+import rootReducer from './rootReducer'
+const store = configureStore({
+  reducer: rootReducer,
+  middleware: [
+    ...customMiddleware,
+    ...getDefaultMiddleware<RootState>(),
+  ] as const,
+})
+// TS2456: Type alias 'RootState' circularly references itself.
+export type RootState = ReturnType<typeof store.getState>
+```
```

Comment:
As of #549, I believe this section could just be removed altogether and this could get merged.

Reply:
```suggestion
```